devito/passes/iet/misc.py

```diff
@@ -75,48 +73,58 @@ def relax_incr_dimensions(iet, **kwargs):
         if root in mapper:
             continue
 
+        # Split iterations to outer and inner
         outer, inner = split(iterations, lambda i: not i.dim.parent.is_Incr)
 
-        # Compute the iteration ranges
-        ranges = []
+        # Get symbolic_max out of each outer dimension
```

Comment:
What makes you think the direction is always ++? 

OK, today in devito the situation is that:

* PARALLEL => TILABLE
* TILABLE <=> IncrDimension
* PARALLEL => Iteration direction FORWARD (++)

hence transitively

* TILABLE => IncrDimension AND Iteration direction FORWARD

so it works, in practice, that you look for `symbolic_max`. But if I were to change the following, even just for fun, from

* PARALLEL => Iteration direction FORWARD (++)

to 

* PARALLEL => Iteration direction BACKWARD (--) 

then this house of cards would collapse.

I'm appreciating this is the way it is in master too, so I can sort of accept to swallow it... but I think it's good to at least place an `assert i.direction is Forward` somewhere ! 

So, yeah, such a long comment for one tiny observation. Apologies

Reply:
:point_up: this is unanswered

Reply:
done